final_task/rss_reader/rss_reader.py

```diff
@@ -0,0 +1,274 @@
+    """Parse the rss"""
+    return feedparser.parse(rss_url)
+
+
+def get_source(parsed: feedparser.FeedParserDict) -> dict:
+    """Returns dictionary with feed information."""
+    feed = parsed['feed']
+    return {
+        'link': feed['link'],
+        'title': feed['title'],
+        'subtitle': feed['subtitle'],
+    }
+
+
+def find_links_in_article(article: feedparser.FeedParserDict) -> list:
+    """Finds links in the article"""
+    image_search = re.compile(r'<\s*img [^>]*src="([^"]+)')
+    img = image_search.findall(article['summary'])
+
+    all_links = []
+    for link in article['links']:
+        if link['href'] not in img:
+            all_links.append(link['href'])
+    all_links += img
+
+    return all_links
+
+
+def unescape(text_to_unescape: str) -> str:
+    """Unescape text"""
+    return html.unescape(text_to_unescape)
+
+
+def get_articles(parsed: feedparser.FeedParserDict, limit: int) -> Tuple[List[single_article.SingleArticle], int]:
+    """Returns list with articles."""
+    articles = []
+    feed = parsed['feed']
+    entries = parsed['entries']
+    logging.info(f'There is {len(entries)} entries')
+
+    no_tags = re.compile('<.*?>')
+
+    if limit is None:
+        limit = len(entries)
+
+    try:
+        for index, entry in enumerate(entries):
+            if index == limit:
+                break
+
+            links_in_article = find_links_in_article(entry)
+
+            if 'published' not in entry:
+                limit -= 1
+                continue
+            articles.append(
+                single_article.SingleArticle(
+                    feed=unescape(feed['title']),
+                    title=unescape(entry['title']),
+                    date=entry['published'],
+                    link=entry['link'],
+                    summary=unescape(re.sub(no_tags, '', entry['summary'])),
+                    links=[f"[{num}]: {link}" for num, link in enumerate(links_in_article, 1)])
+            )
+    except KeyError as value:
+        raise custom_error.ArticleKeyError(f"One of the entries does not have {value} key")
+
+    return articles, limit
+
+
+def print_rss_json(data: List[single_article.SingleArticle]) -> None:
+    """Prints result as JSON in stdout"""
+    print(json.dumps(data, cls=single_article.EnhancedJSONEncoder, ensure_ascii=False))
+
+
+def print_article(data: single_article.SingleArticle) -> None:
+    """Prints a single article in stdout"""
+    print("-" * 120)
+    print(data)
+    print("-" * 120)
+
+
+def print_rss_articles(rss_articles: List[single_article.SingleArticle]) -> None:
+    """Prints source feed and articles"""
+    for article in rss_articles:
+        print_article(article)
+
+
+def check_if_verbose(args: argparse.Namespace) -> None:
+    """Checks if verbose is on and prints logs to stdout"""
+    if args.verbose:
+        with open(LOG_FILE_NAME, 'r') as fin:
+            print(fin.read(), end='')
+        logging.info('Log was printed in stdout')
```

Comment:
If user specified a `--verbose` argument logs should be printed in the process of your app running, not after everything finished working.